src/components/TopBar/HamburgerMenu.jsx

```diff
@@ -1,73 +1,84 @@
-import classnames from 'classnames';
-import {noop} from 'lodash/noop';
+import noop from 'lodash/noop';
 import {t} from 'i18next';
+import tap from 'lodash/tap';
 import PropTypes from 'prop-types';
 import React from 'react';
 import config from '../../config';
+import createMenu, {MenuItem} from './createMenu';
+import HamburgerMenuButton from './HamburgerMenuButton';
 
-export default function HamburgerMenu({
-  isExperimental,
-  isGistExportInProgress,
-  isOpen,
-  isUserAuthenticated,
-  onExportGist,
-  onExportRepo,
-}) {
-  if (!isOpen) {
-    return null;
-  }
+const HamburgerMenu = createMenu({
+  name: 'hamburger',
 
+  renderItems({
+    isExperimental,
+    isGistExportInProgress,
+    isUserAuthenticated,
+    onExportGist,
+    onExportRepo,
+  }) {
+    return tap([], (items) => {
```

Comment:
I'm curious, are you using `tap` here to make it clear from the top that `items` is what we'll be returning? Or is there some other functional purpose?

Reply:
Yeah, maybe because I’ve written a lot of Ruby where it’s a very common pattern, but my instinct is usually to `tap` when the situation is “create some object/data structure, mutate it a bunch, and then return it”. Does it feel awkward to you?